smoke-tests/build.gradle

```diff
@@ -67,7 +67,8 @@ dependencies {
   // Firebase
   implementation "com.google.firebase:firebase-analytics"
   implementation "com.google.firebase:firebase-annotations"
-  implementation "com.google.firebase:firebase-appdistribution:16.0.0-beta02"
+  implementation "com.google.firebase:firebase-appdistribution:16.0.0-beta03"
+  implementation "com.google.firebase:firebase-appdistribution-api:16.0.0-beta03"
```

Comment:
Would you mind explaining the difference between these two artifacts? From reading the source, it seems as though `firbase-appdistribution-api` is utilized for outlining `firbase-appdistribution`, which in turn implements it? I'm assuming it's made that way so that `firbase-appdistribution` can be excluded from release builds for bundle sizes? But even then, it doesn't make sense to me why we have an api variant in the first place.

Reply:
I don't know the details. I'd imagine there may be use cases where some other product which don't require `app-distro`'s implementation, but does need to depend on some class/interface for `app-distro` to recognize and work with it.

@vkryachko Can you share your understanding here?

Reply:
The motivation for the sdk split is exactly that, the idea is to never include `firebase-appdistribution` in prod build: [context](https://firebase.devsite.corp.google.com/docs/app-distribution/set-up-alerts?platform=android)

As for why `-api` exists see how much more complicated setup instructions were before the split: cl/451230288

Reply:
So If I'm understanding this correctly, `firbase-appdistribution-api` acts as a wrapper of sorts around the actual `firebase-appdistribution`. This allows developers to keep "testing logic" in their code for whenever it may be needed in the future, but also allows the developer to exclude the actual testing artifact when not testing (as it not only affects bundle sizes, but is actually against tos for production since it's auto-updating). Essentially, developers can keep testing logic in their code- as the api just hollows it out, versus needing to scrap it everytime they want to release?